src/browser/auth/Login.react.js

```diff
@@ -16,16 +16,14 @@ export default class Login extends Component {
     msg: PropTypes.object.isRequired
   }
 
-  onFormSubmit(e) {
+  async onFormSubmit(e) {
     e.preventDefault();
-    const {actions, auth: {form}} = this.props;
-    actions.login(form.fields)
-      .then(({error, payload: maybeValidationError}) => {
-        if (error)
-          focusInvalidField(this, maybeValidationError);
-        else
-          this.redirectAfterLogin();
-      });
+    const {actions: {login}, auth: {form}} = this.props;
+    const {error, payload: maybeValidationError} = await login(form.fields);
+    if (error)
+      focusInvalidField(this, maybeValidationError);
```

Comment:
Have you tested that if (error) branch? Because to me, it should be:

```js
try {
  await login(form.fields);
} catch (err) {
  focusInvalidField(this, {}); // how to get `maybeValidationError` ? It has to be throwed
}